src/main/scala/Url.scala

```diff
@@ -1,3 +1,98 @@
-object Url {
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.{ContentTypes, HttpEntity, StatusCodes}
+import akka.http.scaladsl.server.Directives._
+import akka.util.Timeout
+import cats.effect.IO
 
+import scala.concurrent.duration._
+import java.sql.Timestamp
+import scala.language.postfixOps
+import spray.json._
+import doobie._
+import doobie.implicits._
+import doobie.implicits.javasql._
+
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
+
+case class UrlIn(userId: Int, value: String, timestamp: String)
+case class UrlOut(userId: Int, value: String, timestamp: Timestamp)
+
+object DatabaseConnection {
+  implicit val cs = IO.contextShift(ExecutionContext.global)
+
+  val xa = Transactor.fromDriverManager[IO](
+    "org.postgresql.Driver", "jdbc:postgresql:browser_history", "postgres", ""
+  )
+
+  def createUrl(userId: Int, timestamp: String, value: String) = {
+    sql"insert into urls(timestamp, value, userid) values (${Timestamp.valueOf(timestamp)}, $value, $userId)"
+      .update
+      .run
+      .transact(xa)
+      .unsafeRunSync()
+  }
+
+  def getAllUrls: List[UrlOut] = {
+    sql"select userid, value, timestamp from urls".query[UrlOut]
+      .to[List]
+      .transact(xa)
+      .unsafeRunSync()
+  }
+}
+
+trait UrlJsonProtocol extends DefaultJsonProtocol {
+  // for json url with timestamp string
+  implicit val urlFormat = jsonFormat3(UrlIn)
+
+  // for url with timestamp Timestamp
+  implicit object TimestampFormat extends JsonFormat[Timestamp] {
+    override def write(obj: Timestamp): JsValue =  JsString(obj.toString)
+
+    override def read(json: JsValue): Timestamp =
+      json match {
+        case JsString(time) =>  Timestamp.valueOf(time)
+        case _ => throw DeserializationException("Expected Timestamp as JsNumber")
+      }
+  }
+  implicit val urlFormatOut = jsonFormat3(UrlOut)
+}
+
+object MainUrls extends App with UrlJsonProtocol with SprayJsonSupport {
```

Comment:
I think it's better to have a separated Main class, where you will add all routes. Please decide & create it after having a conclusion on naming/path (cc @liza-i and @AntociGabriel should also be involved in this discussion).